Firestore/core/src/local/index_manager.h

```diff
@@ -112,6 +112,13 @@ class IndexManager {
   /** Returns all configured field indexes. */
   virtual std::vector<model::FieldIndex> GetFieldIndexes() const = 0;
 
+  /** Cleans up the local indexes records including fast look up map and update
+   * queue. */
+  virtual void CleanUpLocalIndexRecords() = 0;
```

Comment:
Is there a reason this is a dedicated function for C++ but not Java?

Reply:
In Android SDK, this step is be executed as part of `deleteAllIndexes`. Even though in iOS SDK, `deleteAllIndexes` is implemented in persistent class, there are some in memory state needs to be clean up in iOS SDK, so I add this unique function for iOS SDK.

Reply:
Oh I see. Can we make it similar to Android then: have a `deleteAll..` method for index manager and have that clean up both in memory and on disk data?

This should be possible because index manager also hold a pointer to persistence?